www/includes/easyparliament/page.php

```diff
@@ -1287,13 +389,33 @@ public function error_message($message, $fatal = false, $status = 500) {
             );
         }
 
-        $this->message($message, 'error');
+        // if the page has started then we're most likely in an old school page
+        // so we should just print out the error, otherwise stick it in the error
+        // global which will then be displayed by the header template
+        if ( $this->page_started() ) {
+            $this->message($message, 'error');
+        } else {
+            if ( !isset($page_errors) ) {
+                $page_errors = array();
+            }
+            $page_errors[]  = $message;
+        }
 
         if ($fatal) {
+            if (!$this->page_started()) {
+                if (!headers_sent()) {
+                    header("HTTP/1.0 $status Internal Server Error");
+                }
+                $this->page_start();
+            }
+
+            $this->message($message, 'error');
+
             if ($this->within_stripe()) {
                 $this->stripe_end();
             }
             $this->page_end();
+            exit();
```

Comment:
I don't think we want to exit here; all the callers seem to deal with it okay, and this would mean fatal error email wouldn't be sent.